pkg/cmd/bazci/bazci.go

```diff
@@ -57,262 +68,43 @@ func init() {
-			case "go_transition_test":
-				// These tests have their own special testlogs directory.
-				// We can find it by finding the location of the binary
-				// and munging it a bit.
-				args := []string{fullTarget, "-c", compilationMode, "--run_under=realpath"}
-				runOutput, err := runBazelReturningStdout("run", args...)
-				if err != nil {
-					return buildInfo{}, err
-				}
-				var binLocation string
-				for _, line := range strings.Split(runOutput, "\n") {
-					if strings.HasPrefix(line, "/") {
-						// NB: We want the last line in the output that starts with /.
-						binLocation = strings.TrimSpace(line)
-					}
-				}
-				componentsBinLocation := strings.Split(binLocation, "/")
-				componentsTestlogs := strings.Split(testlogsDir, "/")
-				// The second to last component will be the one we need
-				// to replace (it's the output directory for the configuration).
-				componentsTestlogs[len(componentsTestlogs)-2] = componentsBinLocation[len(componentsTestlogs)-2]
-				ret.transitionTests[fullTarget] = strings.Join(componentsTestlogs, "/")
-			case "nodejs_test":
-				ret.tests = append(ret.tests, fullTarget)
-			case "test_suite":
-				// Expand the list of tests from the test suite with another query.
-				allTests, err := runBazelReturningStdout("query", "tests("+fullTarget+")")
-				if err != nil {
-					return buildInfo{}, err
-				}
-				ret.tests = append(ret.tests, strings.Fields(allTests)...)
-			case "_get_x_data":
-				// Can ignore these.
-				continue
-			default:
-				return buildInfo{}, errors.Newf("Got unexpected target kind %v", targetKind)
-			}
-		}
-	}
-
-	return ret, nil
 }
 
 func bazciImpl(cmd *cobra.Command, args []string) error {
-	parsedArgs, err := parseArgs(args, cmd.ArgsLenAtDash())
-	if err != nil {
-		return err
+	if args[0] != buildSubcmd && args[0] != runSubcmd && args[0] != testSubcmd && args[0] != mungeTestXMLSubcmd && args[0] != mergeTestXMLsSubcmd {
+		return errors.Newf("First argument must be `build`, `run`, `test`, `merge-test-xmls`, or `munge-test-xml`; got %v", args[0])
 	}
 
 	// Special case: munge-test-xml/merge-test-xmls don't require running Bazel at all.
 	// Perform the munge then exit immediately.
-	if parsedArgs.subcmd == mungeTestXMLSubcmd {
-		return mungeTestXMLs(*parsedArgs)
+	if args[0] == mungeTestXMLSubcmd {
+		return mungeTestXMLs(args)
 	}
-	if parsedArgs.subcmd == mergeTestXMLsSubcmd {
-		return mergeTestXMLs(*parsedArgs)
+	if args[0] == mergeTestXMLsSubcmd {
+		return mergeTestXMLs(args)
 	}
 
-	info, err := getBuildInfo(*parsedArgs)
+	tmpDir, err := os.MkdirTemp("", "")
 	if err != nil {
 		return err
 	}
+	bepLoc := filepath.Join(tmpDir, "beplog")
+	args = append(args, fmt.Sprintf("--build_event_binary_file=%s", bepLoc))
+	fmt.Println("running bazel w/ args: ", shellescape.QuoteCommand(args))
+	bazelCmd := exec.Command("bazel", args...)
+	bazelCmd.Stdout = os.Stdout
+	bazelCmd.Stderr = os.Stderr
+	err = bazelCmd.Run()
+	// ? What to do with error
```

Comment:
Printing it and continuing LGTM but maybe we should add a warning to the error message that the `bep binary file` may not have been created to guide future debugging?